src/core/handle.ml

```diff
@@ -117,7 +117,7 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
         if !(s.sym_def) <> None then
           fatal pr.pos "Rewriting rules cannot be given for defined \
                         symbol [%s]." s.sym_name;
-        Sr.check_rule ss.builtins r; r
+        Sr.check_rule ss.builtins r
```

Comment:
It feels weird that SR checking should modify the rule. Why is that? I can understand that it could have an effect on the rule (e.g., instantiate some meta variables), but why should it be allowed to make arbitrary modifications by its interface?